app/src/main/java/lt/vilnius/tvarkau/NewProblemActivity.java

```diff
@@ -230,8 +231,16 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                     break;
                 case REQUEST_PLACE_PICKER:
                     Place place = PlacePicker.getPlace(this, data);
-                    mAddProblemLocation.setText(place.getName());
-                    locationCords = place.getLatLng();
+                    CharSequence address = place.getAddress();
+                    String matchVilnius = "Vilnius";
+                    Boolean isVilnius = Arrays.asList(address.toString().split(" ")).contains(matchVilnius);
```

Comment:
Yes, it was a bit hacky. Refactored using Geocoder, seems to be more error prone although not 100% guarantee here as well. :)